diff --git a/third_party/thrift/src/thrift/transport/TSSLSocket.cpp b/third_party/thrift/src/thrift/transport/TSSLSocket.cpp
index 2e1f8e2..004813c 100644
--- a/third_party/thrift/src/thrift/transport/TSSLSocket.cpp
+++ b/third_party/thrift/src/thrift/transport/TSSLSocket.cpp
@@ -149,7 +149,6 @@ void cleanupOpenSSL() {
 #if (OPENSSL_VERSION_NUMBER < OPENSSL_ENGINE_CLEANUP_REQUIRED_BEFORE)
   ENGINE_cleanup();             // https://www.openssl.org/docs/man1.1.0/crypto/ENGINE_cleanup.html - cleanup call is needed before 1.1.0
 #endif
-  CONF_modules_unload(1);
   EVP_cleanup();
   CRYPTO_cleanup_all_ex_data();
   ERR_remove_state(0);
