diff --git a/third_party/absl/BUILD.bazel b/third_party/absl/BUILD.bazel
index c9d4a2d..1917217 100644
--- a/third_party/absl/BUILD.bazel
+++ b/third_party/absl/BUILD.bazel
@@ -41,6 +41,13 @@ config_setting(
     visibility = [":__subpackages__"],
 )
 
+config_setting(
+    name = "windows",
+    constraint_values = [
+        "@bazel_tools//platforms:windows",
+    ],
+)
+
 config_setting(
     name = "osx",
     constraint_values = [
