diff --git a/third_party/absl/copts/GENERATED_copts.bzl b/third_party/absl/copts/GENERATED_copts.bzl
index 6707488..951a091 100644
--- a/third_party/absl/copts/GENERATED_copts.bzl
+++ b/third_party/absl/copts/GENERATED_copts.bzl
@@ -42,7 +42,7 @@ ABSL_GCC_FLAGS = [
     "-Wmissing-declarations",
     "-Woverlength-strings",
     "-Wpointer-arith",
-    "-Wundef",
+    "-Wno-undef",
     "-Wunused-local-typedefs",
     "-Wunused-result",
     "-Wvarargs",
