diff --git a/third_party/glog/src/glog/logging.h.in b/third_party/glog/src/glog/logging.h.in
index 9aac8002..300ae99a 100644
--- a/third_party/glog/src/glog/logging.h.in
+++ b/third_party/glog/src/glog/logging.h.in
@@ -1399,7 +1399,7 @@ class GOOGLE_GLOG_DLL_DECL LogStreamBuf : public std::streambuf {
   }
 
   // This effectively ignores overflow.
-  int_type overflow(int_type ch) {
+  int_type overflow(int_type ch) override {
     return ch;
   }
 
@@ -1927,7 +1927,7 @@ class GOOGLE_GLOG_DLL_DECL NullStreamFatal : public NullStream {
   NullStreamFatal() { }
   NullStreamFatal(const char* file, int line, const CheckOpString& result) :
       NullStream(file, line, result) { }
-  @ac_cv___attribute___noreturn@ ~NullStreamFatal() throw () { _exit(1); }
+  @ac_cv___attribute___noreturn@ ~NullStreamFatal() throw () override { _exit(1); }
 };
 
 // Install a signal handler that will dump signal information and a stack
