diff --git a/folly/logging/BridgeFromGoogleLogging.h b/folly/logging/BridgeFromGoogleLogging.h
index 01e458d9b..9b702a28a 100644
--- a/folly/logging/BridgeFromGoogleLogging.h
+++ b/folly/logging/BridgeFromGoogleLogging.h
@@ -46,7 +46,7 @@ struct BridgeFromGoogleLogging : ::google::LogSink {
       const struct ::tm* pTime,
       const char* message,
       size_t message_len,
-      int32_t usecs);
+      int32_t usecs) override;
 
   void send(
       ::google::LogSeverity severity,
