diff --git a/third_party/protobuf/java/util/BUILD.bazel b/third_party/protobuf/java/util/BUILD.bazel
index ef602075..c791913f 100644
--- a/third_party/protobuf/java/util/BUILD.bazel
+++ b/third_party/protobuf/java/util/BUILD.bazel
@@ -7,9 +7,9 @@ java_library(
     ]),
     visibility = ["//visibility:public"],
     deps = [
-        "//external:error_prone_annotations",
-        "//external:gson",
-        "//external:guava",
+        "@maven//:com_google_errorprone_error_prone_annotations",
+        "@maven//:com_google_code_gson_gson",
+        "@maven//:com_google_guava_guava",
         "//third_party/protobuf/java/core",
         "//third_party/protobuf/java/lite",
     ],
