diff --git a/third_party/protobuf/examples/BUILD b/third_party/protobuf/examples/BUILD
index e9e1e9b1..be062c1e 100644
--- a/third_party/protobuf/examples/BUILD
+++ b/third_party/protobuf/examples/BUILD
@@ -36,12 +36,14 @@ cc_binary(
     name = "add_person_cpp",
     srcs = ["add_person.cc"],
     deps = [":addressbook_cc_proto"],
+    includes = ["."],
 )
 
 cc_binary(
     name = "list_people_cpp",
     srcs = ["list_people.cc"],
     deps = [":addressbook_cc_proto"],
+    includes = ["."],
 )
 
 # Similar to cc_proto_library but for Java.
