diff --git a/bazel/grpc_build_system.bzl b/bazel/grpc_build_system.bzl
index 56b57a1c5..29776c014 100644
--- a/bazel/grpc_build_system.bzl
+++ b/bazel/grpc_build_system.bzl
@@ -55,6 +55,8 @@ def _get_external_deps(external_deps):
             })
         elif dep == "cronet_c_for_grpc":
             ret += ["//third_party/objective_c/Cronet:cronet_c_for_grpc"]
+        elif dep == "madler_zlib":
+            ret += ["@com_github_madler_zlib//:zlib"]
         elif dep.startswith("absl/"):
             ret += ["@com_google_absl//" + dep]
         else:
diff --git a/bazel/grpc_deps.bzl b/bazel/grpc_deps.bzl
index bc66dfeb1..eabef1b9e 100644
--- a/bazel/grpc_deps.bzl
+++ b/bazel/grpc_deps.bzl
@@ -31,11 +31,6 @@ def grpc_deps():
         actual = "@boringssl//:ssl",
     )
 
-    native.bind(
-        name = "madler_zlib",
-        actual = "@zlib//:zlib",
-    )
-
     native.bind(
         name = "protobuf",
         actual = "@com_google_protobuf//:protobuf",
@@ -63,7 +58,7 @@ def grpc_deps():
 
     native.bind(
         name = "gtest",
-        actual = "@com_github_google_googletest//:gtest",
+        actual = "@com_google_googletest//:gtest",
     )
 
     native.bind(
@@ -136,9 +131,9 @@ def grpc_deps():
             url = "https://github.com/google/boringssl/archive/1c2769383f027befac5b75b6cedd25daf3bf4dcf.tar.gz",
         )
 
-    if "zlib" not in native.existing_rules():
+    if "com_github_madler_zlib" not in native.existing_rules():
         http_archive(
-            name = "zlib",
+            name = "com_github_madler_zlib",
             build_file = "@com_github_grpc_grpc//third_party:zlib.BUILD",
             sha256 = "6d4d6640ca3121620995ee255945161821218752b551a1a180f4215f7d124d45",
             strip_prefix = "zlib-cacf7f1d4e3d44d871b605da3b647f07d718623f",
@@ -153,9 +148,9 @@ def grpc_deps():
             url = "https://github.com/google/protobuf/archive/29cd005ce1fe1a8fabf11e325cb13006a6646d59.tar.gz",
         )
 
-    if "com_github_google_googletest" not in native.existing_rules():
+    if "com_google_googletest" not in native.existing_rules():
         http_archive(
-            name = "com_github_google_googletest",
+            name = "com_google_googletest",
             sha256 = "443d383db648ebb8e391382c0ab63263b7091d03197f304390baac10f178a468",
             strip_prefix = "googletest-c9ccac7cb7345901884aabf5d1a786cfa6e2f397",
             url = "https://github.com/google/googletest/archive/c9ccac7cb7345901884aabf5d1a786cfa6e2f397.tar.gz",  # 2019-08-19
