diff --git a/include/openssl/ssl.h b/include/openssl/ssl.h
index e313268..8896b5d 100644
--- a/include/openssl/ssl.h
+++ b/include/openssl/ssl.h
@@ -1200,6 +1200,7 @@ long	SSL_SESSION_get_time(const SSL_SESSION *s);
 long	SSL_SESSION_set_time(SSL_SESSION *s, long t);
 long	SSL_SESSION_get_timeout(const SSL_SESSION *s);
 long	SSL_SESSION_set_timeout(SSL_SESSION *s, long t);
+const char *SSL_SESSION_get0_hostname(const SSL_SESSION *s);
 int	SSL_copy_session_id(SSL *to, const SSL *from);
 X509	*SSL_SESSION_get0_peer(SSL_SESSION *s);
 int	SSL_SESSION_set1_id(SSL_SESSION *s, const unsigned char *sid,
diff --git a/ssl/ssl_sess.c b/ssl/ssl_sess.c
index 44c2e84..accb37f 100644
--- a/ssl/ssl_sess.c
+++ b/ssl/ssl_sess.c
@@ -835,6 +835,13 @@ SSL_SESSION_set_timeout(SSL_SESSION *s, long t)
 	return (1);
 }
 
+const char *
+SSL_SESSION_get0_hostname(const SSL_SESSION *s) {
+	if (s == NULL)
+	    return NULL;
+	return s->tlsext_hostname;
+}
+
 long
 SSL_SESSION_get_timeout(const SSL_SESSION *s)
 {
