diff --git a/BUILD.boost b/BUILD.boost
index d952c3b..dbf82e1 100644
--- a/BUILD.boost
+++ b/BUILD.boost
@@ -446,7 +446,7 @@ cc_library(
     visibility = ["//visibility:public"],
     deps = [
         ":asio",
-        "@openssl//:ssl",
+        "@com_github_hcoona_one//third_party/boringssl:ssl",
     ],
 )

@@ -1060,10 +1060,10 @@ boost_library(
         ":type",
         ":type_traits",
         ":utility",
-        "@com_github_facebook_zstd//:zstd",
-        "@net_zlib_zlib//:zlib",
-        "@org_bzip_bzip2//:bz2lib",
-        "@org_lzma_lzma//:lzma",
+        "@com_github_hcoona_one//third_party/zstd",
+        "@com_github_hcoona_one//third_party/zlib",
+        "@com_github_hcoona_one//third_party/bzip2:bz2",
+        "@com_github_hcoona_one//third_party/xz:lzma",
     ],
 )

