diff --git a/BUILD b/BUILD
index efc3d8e7f..1ec5ec25a 100644
--- a/BUILD
+++ b/BUILD
@@ -24,7 +24,7 @@ config_setting(
 # ZLIB configuration
 ################################################################################
 
-ZLIB_DEPS = ["@zlib//:zlib"]
+ZLIB_DEPS = ["@com_github_madler_zlib//:zlib"]
 
 ################################################################################
 # Protobuf Runtime Library
diff --git a/WORKSPACE b/WORKSPACE
index 35ef1d583..37c336b4b 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -5,11 +5,6 @@ local_repository(
     path = "examples",
 )
 
-local_repository(
-    name = "submodule_gmock",
-    path = "third_party/googletest",
-)
-
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 load("//:protobuf_deps.bzl", "protobuf_deps")
 
@@ -23,12 +18,12 @@ bind(
 
 bind(
     name = "gtest",
-    actual = "@submodule_gmock//:gtest",
+    actual = "@com_google_googletest//:gtest",
 )
 
 bind(
     name = "gtest_main",
-    actual = "@submodule_gmock//:gtest_main",
+    actual = "@com_google_googletest//:gtest_main",
 )
 
 maven_jar(
diff --git a/protobuf_deps.bzl b/protobuf_deps.bzl
index 9209b1c5a..a4377cd58 100644
--- a/protobuf_deps.bzl
+++ b/protobuf_deps.bzl
@@ -13,9 +13,9 @@ def protobuf_deps():
             urls = ["https://github.com/bazelbuild/bazel-skylib/archive/2169ae1c374aab4a09aa90e65efe1a3aad4e279b.tar.gz"],
         )
 
-    if not native.existing_rule("zlib"):
+    if not native.existing_rule("com_github_madler_zlib"):
         http_archive(
-            name = "zlib",
+            name = "com_github_madler_zlib",
             build_file = "@com_google_protobuf//:third_party/zlib.BUILD",
             sha256 = "629380c90a77b964d896ed37163f5c3a34f6e6d897311f1df2a7016355c45eff",
             strip_prefix = "zlib-1.2.11",
