diff --git a/BUILD b/BUILD
index 4c9b57a..8ca0c5e 100644
--- a/BUILD
+++ b/BUILD
@@ -55,8 +55,11 @@ STRICT_C_OPTIONS = select({
         "-Wmissing-declarations",
         "-Wmissing-prototypes",
         "-Wno-strict-aliasing",
+        "-fno-strict-aliasing",
         "-Wshadow",
         "-Wsign-compare",
+        "-Wno-implicit-int-float-conversion",
+        "-D_POSIX_C_SOURCE",
     ],
 })
 
