diff --git a/third_party/snappy/snappy_test_data.cc b/third_party/snappy/snappy_test_data.cc
index 3ed532d..bde557e 100644
--- a/third_party/snappy/snappy_test_data.cc
+++ b/third_party/snappy/snappy_test_data.cc
@@ -34,20 +34,23 @@
 #include <cstdlib>
 #include <string>

+#include "tools/cpp/runfiles/runfiles.h"
 #include "third_party/snappy/snappy-test.h"

 namespace snappy {

 std::string ReadTestDataFile(const char* base, size_t size_limit) {
-  std::string srcdir;
-  const char* srcdir_env = std::getenv("srcdir");  // This is set by Automake.
-  if (srcdir_env) {
-    srcdir = std::string(srcdir_env) + "/";
-  }
+  constexpr char kSrcDir[] = "com_github_hcoona_one/third_party/snappy/";
+  using bazel::tools::cpp::runfiles::Runfiles;
+
+  std::string error;
+  std::unique_ptr<Runfiles> runfiles{Runfiles::CreateForTest(&error)};
+  CHECK(runfiles);

   std::string contents;
-  CHECK_OK(file::GetContents(srcdir + "testdata/" + base, &contents,
-                             file::Defaults()));
+  CHECK_OK(file::GetContents(
+      runfiles->Rlocation(std::string(kSrcDir) + "testdata/" + base), &contents,
+      file::Defaults()));
   if (size_limit > 0) {
     contents = contents.substr(0, size_limit);
   }
