diff --git a/src/examples/cpp/console.cpp b/src/examples/cpp/console.cpp
index 6a01d8a..3c8d133 100644
--- a/src/examples/cpp/console.cpp
+++ b/src/examples/cpp/console.cpp
@@ -16,12 +16,19 @@
  */
 
 #include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
 #include <log4cxx/logger.h>
 #include <log4cxx/consoleappender.h>
 #include <log4cxx/simplelayout.h>
 #include <log4cxx/logmanager.h>
 #include <iostream>
 #include <locale.h>
+#include <cstring>
+#include <cstdio>
+#ifndef WIN32
+#include <stdint.h>
+#endif
 
 using namespace log4cxx;
 using namespace log4cxx::helpers;
diff --git a/src/main/cpp/inputstreamreader.cpp b/src/main/cpp/inputstreamreader.cpp
index 52b1c0a..beea2be 100644
--- a/src/main/cpp/inputstreamreader.cpp
+++ b/src/main/cpp/inputstreamreader.cpp
@@ -21,6 +21,9 @@
 #include <log4cxx/helpers/pool.h>
 #include <log4cxx/helpers/bytebuffer.h>
 
+#include <cstdio>
+#include <cstring>
+
 using namespace log4cxx;
 using namespace log4cxx::helpers;
 
diff --git a/src/main/cpp/socketoutputstream.cpp b/src/main/cpp/socketoutputstream.cpp
index 185f835..7c1e449 100644
--- a/src/main/cpp/socketoutputstream.cpp
+++ b/src/main/cpp/socketoutputstream.cpp
@@ -20,6 +20,9 @@
 #include <log4cxx/helpers/socket.h>
 #include <log4cxx/helpers/bytebuffer.h>
 
+#include <cstdio>
+#include <cstring>
+
 using namespace log4cxx;
 using namespace log4cxx::helpers;
 
