diff --git a/third_party/arrow/src/arrow/util/logging.cc b/third_party/arrow/src/arrow/util/logging.cc
index 029b277..b3bc40a 100644
--- a/third_party/arrow/src/arrow/util/logging.cc
+++ b/third_party/arrow/src/arrow/util/logging.cc
@@ -28,6 +28,16 @@
 #include <signal.h>
 #include <vector>
 
+// Undefine our version before include glog to avoid macro redefined warning.
+#undef DCHECK
+#undef DCHECK_OK
+#undef DCHECK_EQ
+#undef DCHECK_NE
+#undef DCHECK_LE
+#undef DCHECK_LT
+#undef DCHECK_GE
+#undef DCHECK_GT
+
 #include "glog/logging.h"
 
 // Restore our versions of DCHECK and friends, as GLog defines its own
