diff --git a/test/msgpack_test.cpp b/test/msgpack_test.cpp
index fb35c683..5a56386f 100644
--- a/test/msgpack_test.cpp
+++ b/test/msgpack_test.cpp
@@ -170,10 +170,10 @@ TEST(MSGPACK, simple_buffer_float)
     float val2;
     obj.convert(&val2);

-    if (isnan(val1))
-      EXPECT_TRUE(isnan(val2));
-    else if (isinf(val1))
-      EXPECT_TRUE(isinf(val2));
+    if (std::isnan(val1))
+      EXPECT_TRUE(std::isnan(val2));
+    else if (std::isinf(val1))
+      EXPECT_TRUE(std::isinf(val2));
     else
       EXPECT_TRUE(fabs(val2 - val1) <= kEPS);
   }
@@ -260,10 +260,10 @@ TEST(MSGPACK, simple_buffer_double)
     double val2;
     obj.convert(&val2);

-    if (isnan(val1))
-      EXPECT_TRUE(isnan(val2));
-    else if (isinf(val1))
-      EXPECT_TRUE(isinf(val2));
+    if (std::isnan(val1))
+      EXPECT_TRUE(std::isnan(val2));
+    else if (std::isinf(val1))
+      EXPECT_TRUE(std::isinf(val2));
     else
       EXPECT_TRUE(fabs(val2 - val1) <= kEPS);
   }
