diff --git a/parquet/statistics.cc b/parquet/statistics.cc
index 6c6cefe..728e03c 100644
--- a/parquet/statistics.cc
+++ b/parquet/statistics.cc
@@ -222,11 +222,11 @@ CleanStatistic(std::pair<T, T> min_max) {
 
   // Ignore if one of the value is nan.
   if (std::isnan(min) || std::isnan(max)) {
-    return nonstd::nullopt;
+    return arrow::util::nullopt;
   }
 
   if (min == std::numeric_limits<T>::max() && max == std::numeric_limits<T>::lowest()) {
-    return nonstd::nullopt;
+    return arrow::util::nullopt;
   }
 
   T zero{};
@@ -244,7 +244,7 @@ CleanStatistic(std::pair<T, T> min_max) {
 
 optional<std::pair<FLBA, FLBA>> CleanStatistic(std::pair<FLBA, FLBA> min_max) {
   if (min_max.first.ptr == nullptr || min_max.second.ptr == nullptr) {
-    return nonstd::nullopt;
+    return arrow::util::nullopt;
   }
   return min_max;
 }
@@ -252,7 +252,7 @@ optional<std::pair<FLBA, FLBA>> CleanStatistic(std::pair<FLBA, FLBA> min_max) {
 optional<std::pair<ByteArray, ByteArray>> CleanStatistic(
     std::pair<ByteArray, ByteArray> min_max) {
   if (min_max.first.ptr == nullptr || min_max.second.ptr == nullptr) {
-    return nonstd::nullopt;
+    return arrow::util::nullopt;
   }
   return min_max;
 }
